controller/eventGroupController.js

```diff
@@ -0,0 +1,172 @@
+    const eventGroup = new EventGroup(req.body);
+
+    if (loggedUser.companyID != eventGroup.companyID) {
+        return res.status(HttpStatus.UNAUTHORIZED).json({
+            status: StatusEnum['ERROR'],
+            message: 'Unauthorized'
+        });
+    }
+
+    // Check if event exists 
+    Event.findOne({ _id: eventGroup.eventID, companyID: eventGroup.companyID }).then(event => {
+        if (!event) {
+            return res.status(HttpStatus.NOT_FOUND).json({
+                status: StatusEnum['ERROR'],
+                message: 'Event not found'
+            });
+        } else {
+            eventGroup.save().then(() => {
+                return res.status(HttpStatus.CREATED).json(eventGroup);
+            }).catch(err => {
+                return res.status(HttpStatus.INTERNAL_SERVER_ERROR).json({
+                    status: StatusEnum['ERROR'],
+                    message: ValidationUtil.buildErrorMessage(err, 'create', 'eventGroup')
+                });
+            });
+        }
+    }).catch(err => {
+        return res.status(HttpStatus.INTERNAL_SERVER_ERROR).json({
+            status: StatusEnum['ERROR'],
+            message: ValidationUtil.buildErrorMessage(err, 'create', 'eventGroup')
+        });
+    });
+};
+
+const retrieveAll = (req, res) => {
+    const loggedUser = req.decoded;
+
+    EventGroup.find({ companyID: loggedUser.companyID }).then(eventGroups => {
+        if (eventGroups) {
+            return res.status(HttpStatus.OK).json(eventGroups);
+        } else {
+            return res.status(HttpStatus.NOT_FOUND).json({
+                status: StatusEnum['ERROR'],
+                message: 'Event groups not found.'
+            });
+        }
+    }).catch(err => {
+        return res.status(HttpStatus.INTERNAL_SERVER_ERROR).json({
+            status: StatusEnum['ERROR'],
+            message: ValidationUtil.buildErrorMessage(err, 'retrieveAll', 'eventGroup')
+        });
+    });
+}
+
+const retrieveById = (req, res) => {
+    const eventGroupId = req.params.id;
+    const loggedUser = req.decoded;
+
+    if (!ValidationUtil.isValidObjectId(eventGroupId)) {
+        return res.status(HttpStatus.BAD_REQUEST).json({
+            status: 'Error',
+            message: 'Invalid ID'
+        });
+    }
+
+    EventGroup.find({ _id: eventGroupId, companyID: loggedUser.companyID }).then(eventGroups => {
+        if (eventGroups) {
+            return res.status(HttpStatus.OK).json(eventGroups);
+        } else {
+            return res.status(HttpStatus.NOT_FOUND).json({
+                status: StatusEnum['ERROR'],
+                message: 'Event group not found.'
+            });
+        }
+    }).catch(err => {
+        return res.status(HttpStatus.INTERNAL_SERVER_ERROR).json({
+            status: StatusEnum['ERROR'],
+            message: ValidationUtil.buildErrorMessage(err, 'retrieveAll', 'eventGroup')
+        });
+    });
+};
+
+const update = (req, res) => {
```

Comment:
I'm guessing for this function, since only the Super-Admin/Admin users can update, that check will be done in that special Auth middleware function and that that is the reason it is left out here?

Reply:
Yes